src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -1095,6 +1095,113 @@ public function findServicesByServiceGroups(array $serviceGroups): array
         return $servicesByServiceGroupId;
     }
 
+    /**
+     * @param int $hostId
+     * @param int $serviceId
+     * @return array|null
+     */
+    public function findServiceGroupsByHostAndService(int $hostId, int $serviceId): ?array
+    {
+
+        $serviceGroups = [];
+
+        if ($this->hasNotEnoughRightsToContinue()) {
+            return $serviceGroups;
+        }
+
+        $sqlExtraParameters = [];
+        $subRequest = '';
+        if (!$this->isAdmin()) {
+            $sqlExtraParameters = [':contact_id' => [\PDO::PARAM_INT => $this->contact->getId()]];
+
+            // Not an admin, we must to filter on contact
+            $subRequest .=
+                ' INNER JOIN `:db`.acl_resources_sg_relations sgr
```

Comment:
you should also check if centeon.acl_resources.all_servicegroups === '1' , then the user can see all servicegroups (even if there is no relation in centreon.acl_resources_sg_relations)

Reply:
![image](https://user-images.githubusercontent.com/11978823/74249242-292b0e80-4ce9-11ea-891d-4d63f19445b2.png)

Reply:
thanks! will check that